test/cutting/cut_finding/test_best_first_search.py

```diff
@@ -0,0 +1,126 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" Tests for best_first_search module."""
+
+from __future__ import annotations
+
+from pytest import fixture
+from numpy import inf
+from circuit_knitting.cutting.cut_finding.circuit_interface import (
+    SimpleGateList,
+    CircuitElement,
+    GateSpec,
+)
+from circuit_knitting.cutting.cut_finding.cut_optimization import CutOptimization
+from circuit_knitting.cutting.cut_finding.optimization_settings import (
+    OptimizationSettings,
+)
+from circuit_knitting.cutting.find_cuts import DeviceConstraints
```

Comment:
My guess is that this first line is doing to break CI.  Maybe there are similar things elsewhere?  The below would be better probably (but not sure if black will like it)
```suggestion
from circuit_knitting.cutting.find_cuts import DeviceConstraints, OptimizationSettings
```